cloud/endagaweb/views/network.py

```diff
@@ -453,3 +455,269 @@ def get(self, request, network_id):
+        return redirect(urlresolvers.reverse('network-denominations'))
+
+    def delete(self, request):
+        """Handles delete requests."""
+        response = {
+            'status': 'ok',
+            'messages': [],
+        }
+        dnm_id = request.GET.get('id') or False
+        if dnm_id:
+            try:
+                denom = models.NetworkDenomination.objects.get(id=dnm_id)
+                denom.delete()
+                response['status'] = 'success'
+                messages.success(request,
+                                 'Denomination deleted successfully.',
+                                 extra_tags='alert alert-success')
+            except models.NetworkDenomination.DoesNotExist:
+                response['status'] = 'failed'
+                messages.error(
+                    request, 'Invalid denomination ID.',
+                    extra_tags='alert alert-danger')
+        else:
+            response['status'] = 'failed'
+            messages.error(
+                request, 'Invalid request data.',
+                extra_tags='alert alert-danger')
+        return http.HttpResponse(json.dumps(response),
+                                 content_type="application/json")
+
+
+class NetworkBalanceLimit(ProtectedView):
+    """Edit basic network info (to add credit to Network)."""
+
+    def get(self, request):
+        """Handles GET requests."""
+        user_profile = models.UserProfile.objects.get(user=request.user)
+        network = user_profile.network
+        # Set the context with various stats.
+        currency = network.subscriber_currency
+        context = {
+            'networks': get_objects_for_user(request.user, 'view_network',
+                                             klass=models.Network),
+            'user_profile': user_profile,
+            'network': network,
+            'currency': CURRENCIES[network.subscriber_currency],
+            'network_balance_limit_form': dashboard_forms.NetworkBalanceLimit({
+                'limit': '',
+                'transitions': '',
+
+            }),
+        }
+        # Render template.
+        edit_template = template.loader.get_template(
+            'dashboard/network_detail/network-balancelimit.html')
+        html = edit_template.render(context, request)
+        return http.HttpResponse(html)
+
+    def post(self, request):
+        """Handles POST requests."""
+        user_profile = models.UserProfile.objects.get(user=request.user)
+        network = user_profile.network
+        success = []
+        if 'limit' not in request.POST:
+            return http.HttpResponseBadRequest()
+        if 'transaction' not in request.POST:
+            return http.HttpResponseBadRequest()
+        limit = float(request.POST.get('limit') or 0)
+
+        if request.POST.get('transaction') == "" and request.POST.get(
+                'limit') == "":
+            error_text = 'Error : please provide value.'
```

Comment:
Just saying you can do a lot of these checks in subclasses of `forms.Form` and raise `forms.ValidationError`